.github/actions/compare-regex-versions/action.yaml

```
@@ -0,0 +1,67 @@
+name: Compare Regex Versions
+description: Extract and compare version strings from two files using regex patterns
+
+inputs:
+  file1:
+    description: "Path to the first file"
+    required: true
+  file1-regex:
+    description: 'Perl-compatible regex pattern to extract version from file1 (use \K and lookahead for capture)'
+    required: true
+  file1-label:
+    description: "Human-readable label for file1 (for error messages)"
+    required: true
+
+  file2:
+    description: "Path to the second file"
+    required: true
+  file2-regex:
+    description: 'Perl-compatible regex pattern to extract version from file2 (use \K and lookahead for capture)'
+    required: true
+  file2-label:
+    description: "Human-readable label for file2 (for error messages)"
+    required: true
+
+  error-message:
+    description: "Custom error message to display if versions do not match (optional, uses default if not provided)"
+    required: false
+    default: ""
+
+runs:
+  using: composite
```

```suggestion
  using: "composite"
```
Nit: for consistency as all the existing actions use quotes here (although without quotes looks better)